service/share/share.go

```diff
@@ -145,16 +146,52 @@ func (s *service) GetSharesByNamespace(ctx context.Context, root *Root, nID name
 	}
 
 	namespacedShares := make([]Share, 0)
+	nodeCh := make(chan []format.Node)
+	errCh := make(chan error)
+	wg := sync.WaitGroup{}
+	newCtx, cancel := context.WithCancel(ctx)
+
```

Comment:
It's ok to have multiple defers in a func and it is a good practice to keep each defer next to the value a defer is closing/canceling/flushing or etc.

Reply:
reworked